src/dataframe.rs

```diff
@@ -212,6 +212,39 @@ impl PyDataFrame {
         Ok(Self::new(df))
     }
 
+    /// Thin wrapper for datafusion-rust `join_on`
+    /// Slightly modified from original `join` above.
+    fn join_on(
+            &self,
+            right: PyDataFrame,
+            on_exprs: Vec<PyExpr>,
+            how: &str,
+        ) -> PyResult<Self> {
+        let join_type = match how {
+            "inner" => JoinType::Inner,
+            "left" => JoinType::Left,
+            "right" => JoinType::Right,
+            "full" => JoinType::Full,
+            "semi" => JoinType::LeftSemi,
+            "anti" => JoinType::LeftAnti,
+            how => {
+                return Err(DataFusionError::Common(format!(
+                    "The join type {how} does not exist or is not implemented"
+                ))
+                .into());
+            }
```

Comment:
shouldn't this be covered by the https://doc.rust-lang.org/std/str/trait.FromStr.html impl for the enum?

Reply:
I think so, but the trait doesn't appear to implemented https://github.com/apache/arrow-datafusion/blob/main/datafusion/expr/src/logical_plan/plan.rs#L1115-L1156, so I copied the approach from `join` in this same file for now.

Should I add it to the main repo? Should it be a blocker for this change here?

I'm new to this community so please bear with my questions 😬!